### PR TITLE
Remove allowedAudienceForAssociatedRoles from additionalSpProperties list

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
### Proposed changes in this pull request

remove allowedAudienceForAssociatedRoles from additionalSpProperties list

### Related Issue
- https://github.com/wso2/product-is/issues/17536

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 1]
```diff
@@ -2136,6 +2136,14 @@ public ServiceProvider getApplication(int applicationId) throws IdentityApplicat
         }
     }
 
+    private  ServiceProviderProperty[] getSpProperties(List<ServiceProviderProperty> propertyList) {
```
Comment (AnuradhaSK) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:2139]: handle this similar to isManagementApp property handling

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 2]
```diff
@@ -2234,7 +2242,7 @@ public ServiceProvider getApplicationWithRequiredAttributes(int applicationId, L
                     if (ADVANCED_CONFIG.equals(requiredAttribute)) {
                         readAndSetConfigurationsFromProperties(propertyList,
                                 serviceProvider.getLocalAndOutBoundAuthenticationConfig());
-                        serviceProvider.setSpProperties(propertyList.toArray(new ServiceProviderProperty[0]));
+                        serviceProvider.setSpProperties((getSpProperties(propertyList)));
```
Comment (sadilchamishka) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:2245]:
```suggestion
                        serviceProvider.setSpProperties(getSpProperties(propertyList));
```

Comment (sadilchamishka) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:2245]: Is there additional brackets?

**Comment by ChanikaRuchini** [comment]
fixed with https://github.com/wso2/identity-api-server/pull/527

**Pull request closed without merging**